src/inference/hmc.jl

```diff
@@ -33,47 +50,107 @@ sample(gdemo([1.5, 2]), HMC(1000, 0.01, 10))
 ```
 """
 mutable struct HMC{AD, space, metricT <: AHMC.AbstractMetric} <: StaticHamiltonian{AD}
-    n_iters     ::  Int       # number of samples
     ϵ           ::  Float64   # leapfrog step size
     n_leapfrog  ::  Int       # leapfrog step number
 end
+
+transition_type(::Sampler{<:Hamiltonian}) = Transition
+alg_str(::Sampler{<:Hamiltonian}) = "HMC"
+
 HMC(args...) = HMC{ADBackend()}(args...)
-function HMC{AD}(n_iters::Int, ϵ::Float64, n_leapfrog::Int, ::Type{metricT}, space::Tuple) where {AD, metricT <: AHMC.AbstractMetric}
-    return HMC{AD, space, metricT}(n_iters, ϵ, n_leapfrog)
+function HMC{AD}( ϵ::Float64, n_leapfrog::Int, ::Type{metricT}, space::Tuple) where {AD, metricT <: AHMC.AbstractMetric}
+    return HMC{AD, space, metricT}(ϵ, n_leapfrog)
 end
 function HMC{AD}(
-    n_iters::Int,
     ϵ::Float64,
     n_leapfrog::Int,
     ::Tuple{};
     kwargs...
 ) where AD
-    return HMC{AD}(n_iters, ϵ, n_leapfrog; kwargs...)
+    return HMC{AD}(ϵ, n_leapfrog; kwargs...)
 end
 function HMC{AD}(
-    n_iters::Int,
     ϵ::Float64,
     n_leapfrog::Int,
     space::Symbol...;
     metricT=AHMC.UnitEuclideanMetric
 ) where AD
-    return HMC{AD}(n_iters, ϵ, n_leapfrog, metricT, space)
+    return HMC{AD}(ϵ, n_leapfrog, metricT, space)
+end
+
+function sample_init!(
+    ::AbstractRNG,
+    model::Model,
+    spl::Sampler{<:Hamiltonian},
+    N::Integer;
+    adaptor=AHMCAdaptor(spl.alg),
+    init_theta::Union{Nothing,Array{<:Any,1}}=nothing,
+    rng::AbstractRNG=GLOBAL_RNG,
+    discard_adapt::Bool=true,
+    verbose::Bool=true,
+    resume_from=nothing,
+    kwargs...
+)
+    # Resume the sampler.
+    set_resume!(spl; kwargs...)
+
+    # Get `init_theta`
+    if init_theta != nothing
+        verbose && @info "Using passed-in initial variable values" init_theta
+        # Convert individual numbers to length 1 vector; `ismissing(v)` is needed as `size(missing)` is undefined`
+        init_theta = [ismissing(v) || size(v) == () ? [v] : v for v in init_theta]
```

Comment:
Do you think we can move this to the shared interface?

Reply:
That's a great idea, I'll stick this in `InterfaceExtensions.jl` and enable it for all the samplers.